### PR TITLE
Migrate to 'go mod' for tracking dependencies

### PR DESCRIPTION
**Pull request by alfrunes**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module github.com/mendersoftware/go-lib-micro
+
+go 1.14
+
+require (
+	github.com/ant0ine/go-json-rest v3.3.2+incompatible
```
Comment (merlin-northern) [inline go.mod:6]: this is ce4b71cdf7ba29ef443d704bd923f5bbf281ee30 (Wed Nov 30 19:48:30 2016 -0800), but what we use in go-lib-micro is ebb33769ae013bd5f518a8bac348c310dea768b8 (Wed Sep 13 04:12:08 2017 +0000) why the change?

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	github.com/mendersoftware/inventory v0.0.0-20200522065057-a25956b1ec7b
```
Comment (merlin-northern) [inline go.mod:7]: what we currently use is `a1bca36b3cff2262d5bc9347a9af06b90ae55de5` (_Tue Dec 3 14:02:17 2019 +0100_) this is latest commit, so, maybe lets bind it to `2.0.0b1-build2` inventory git tag? just suggestions and nitpicking. I understand that this upgrade is intentional?

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+	github.com/mendersoftware/mendertesting v0.0.0-20200519115418-fd4090fcab1a
```
Comment (merlin-northern) [inline go.mod:8]: ok, upgrade of mendertesting is required, right?

File: go.mod (hunk 1 above)
Comment (alfrunes) [reply go.mod:6]: This is just the auto-filled version (by go mod init), which defaults to the latest recognized semver tag. I suppose I can update this to the latest commit on master.

File: go.mod (hunk 3 above)
Comment (alfrunes) [reply go.mod:8]: I suppose it doesn't hurt :)

File: go.mod (hunk 2 above)
Comment (alfrunes) [reply go.mod:7]: >  maybe lets bind it to 2.0.0b1-build2 inventory git tag? 

This is actually the commit hash for that tag (unfortunately the tag is not recognized as a semver tag, hence the v0.0.0).
